server/src/main/java/org/apache/iotdb/db/engine/compaction/cross/inplace/manage/CrossSpaceMergeResource.java

```diff
@@ -103,12 +103,15 @@ public void clear() throws IOException {
     fileReaderCache.clear();
     fileWriterCache.clear();
     modificationCache.clear();
-    measurementSchemaMap.clear();
     chunkWriterCache.clear();
   }
 
   public IMeasurementSchema getSchema(PartialPath path) {
-    return measurementSchemaMap.get(path);
+    try {
+      return IoTDB.metaManager.getMeasurementMNode(path).getSchema();
+    } catch (MetadataException e) {
+      return null;
+    }
```

Comment:
Using methods like getSeriesSchema in MManager to get schema directly

Reply:
done